src/k8s/pkg/utils/certificate.go

```diff
@@ -26,3 +32,75 @@ func SplitIPAndDNSSANs(extraSANs []string) ([]net.IP, []string) {
 
 	return ipSANs, dnsSANs
 }
+
+func TLSClientConfig(remoteCert *x509.Certificate) (*tls.Config, error) {
```

Comment:
please add a `// TLSClientConfig returns ...`, as `utils.TLSClientConfig` is a very generic name.

Also, I would recommend naming this something like: `TLSClientConfigWithTrustedCertificate()`